lib/py/src/protocol/TProtocol.py

```diff
@@ -118,6 +118,8 @@ def writeDouble(self, dub):
         pass
 
     def writeString(self, str_val):
+        if isinstance(str_val, bytes):
+            str_val = str_val.decode('utf8')
```

Comment:
This would break py2

Reply:
It does, yes.  And its absence breaks py3.6.  I'm working on a fix for this.

Reply:
Can `six` help here?  `six.binary_type` for example.  Or check for python version and use `bytearray` for py2 and bytes for py3?